### ISSUE TITLE
Logic App 'Create a new watchlist with data (Raw content)' returns 400

### ISSUE THREAD

**Issue by d3vzer0** [Azure/Azure-Sentinel#5040]
**Describe the bug**
The create/update Watchlist API endpoint returns a 400 (invalid CSV content) when trying to create a new Sentinel Watchlist using the built-in Logic Apps connector. The error returned is:

```
"Content": "{\"error\":{\"code\":\"400\",\"message\":\"Invalid csv content : no watchlist items found in the provided rawContent <rawContent>!\"}}",
```

**To Reproduce**
1. Create a new Logic App
2. Add a 'Watchlist - Create a new watchlist with data (Raw content)' task
3. Fill in the necessary subscription, workspace etc
4. In the 'Specify Watchlist fields', fill in the default API example body (https://docs.microsoft.com/nl-nl/rest/api/securityinsights/preview/watchlists/create-or-update):

```
{
  "contentType": "text/csv",
  "description": "Watchlist from CSV content",
  "displayName": "High Value Assets Watchlist",
  "itemsSearchKey": "header1",
  "numberOfLinesToSkip": 1,
  "provider": "Microsoft",
  "rawContent": "This line will be skipped\nheader1,header2\nvalue1,value2",
  "source": "Local file"
}
```

**Expected behavior**
A new watchlist should be created with the sample content. Replacing the rawContent newline with `\r\n` produces the same error.

**Comment by d3vzer0** [comment]
> Try adding **\r** like this:
> 
> `"rawContent": "This line will be skipped\r\nheader1,header2\r\nvalue1,value2",`

Thanks for the suggestion :) I already tried to replace the \n newlines with \r\n but didn't make any difference :( I used the literal example from Microsofts own blogpost so no idea what the difference could be 🤔 Maybe something going wrong with the Logic app parsing literal newlines? Also tried using a payload/json body as generated when manually using the Sentinel UI to create a watchlist (ie. copied the json body using Firefox network inspection).

**Comment by napieram** [comment]
I just copied the properties from the template and it still threw the same error.

`{
  "StatusCode": "BadRequest",
  "ReasonPhrase": "Bad Request",
  "Content": "{\"error\":{\"code\":\"400\",\"message\":\"Invalid csv content : no watchlist items found in the provided rawContent <rawContent>!\"}}",
  "Headers": {
    "Pragma": [`

**Comment by napieram** [comment]
Is there a way to check what api version the logic app is using? I tried to replicate this using REST api. Creating a watchlist with items does not appear to be working since [2022-01-01-preview](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-01-01-preview).

**Comment by BenjiSec** [comment]
Can you please try to do it, but without "source" part?

**Comment by napieram** [comment]
Without source
`{
  "error": {
    "code": 500,
    "source": "logic-apis-eastus.azure-apim.net",
    "clientRequestId": "*****",
    "message": "BadGateway",
    "innerError": {
      "status": 500,
      "message": "The given key was not present in the dictionary.\r\nclientRequestId: *****",
      "error": {
        "message": "The given key was not present in the dictionary."
      },
      "source": "azuresentinel-eus.azconn-eus-002.p.azurewebsites.net"
    }
  }
}`

**Comment by pagoudjo** [comment]
Taking a look at it also.

**Comment by BenjiSec** [comment]
@napieram, @d3vzer0 - I got the same yesterday and we are working in background to find out more. @pagoudjo will keep us updated.

**Comment by d3vzer0** [comment]
Managed to get it working with the help from Yaser (Microsoft) :) 

The trick is to include the header of the CSV but keep NumberOfLinesToSkip = 1, ie.
```
{
  "contentType": "text/csv",
  "description": "Watchlist from CSV content",
  "displayName": "High Value Assets Watchlist",
  "itemsSearchKey": "header1",
  "numberOfLinesToSkip": 1,
  "provider": "Microsoft",
  "rawContent": "header1,header2\r\nvalue1,value2",
  "source": "Local file"
}
```

The header is still parsed correctly though! So its not being skipped which may seem confusing by the name numberOfLinesToSkip

**Comment by pagoudjo** [comment]
Hello,
Thank you for letting us know.

Indeed, the header always has to be included in the raw data when using the action.

Cheers,
Patrick

________________________________

**Comment by d3vzer0** [comment]
@pagoudjo Indeed, but the numberOfLinesToSkip is confusing. It makes the assumption to actually skip the header while it is essentially part of the CSV. The example is incorrect since it essentially skips the dummy line and starts at the header instead of skipping it. Maybe the docs have to be updated to represent this :)

**Comment by pagoudjo** [comment]
Joey, the numberOfLinesToSkip  (NLS) indicates the number of lines to skip before we find the headers line. Let say you set it to N, then you header line will be considered to be the one at line N+1.

Therefore in your example, if the first line is the dummy line and you set NLS to 1, it will read your headers at line 2.

I hope this helps.

Cheers,
Patrick

________________________________

**Comment by d3vzer0** [comment]
@pagoudjo That was my assumption as well, but it only works when you skip the header ie. the header is on the first line and numberOfLinesToSkip=1. Hence the default example doesn't work :(

**Comment by pagoudjo** [comment]
@d3vzer0 Let me take another look as this might be a "different scenario". Getting back to you shortly.

**Comment by pagoudjo** [comment]
@d3vzer0  FYI specifying "contentType": "text/csv" is not required for this action.

**Comment by pagoudjo** [comment]
@d3vzer0 Once again I think the numberOfLinesToSkip skips some lines before finding the Headers line. Its possible you had more lines than necessary in your raw data.

**Comment by d3vzer0** [comment]
@pagoudjo Thanks! I removed those fields but as mentioned earlier and confirmed by @napieram as well, adding numberOfLinesToSkip=1 solves the issue since it doesn't actually skip the header. I believe the docs/example needs to be updated.

@napieram Yep that solves the issue :) It doesn't skip the first line (header), the documentation/default example is incorrect.

**Comment by pagoudjo** [comment]
Thank you @d3vzer0 and @napieram!
We'll also keep you updated on upcoming changes to this area.

**Comment by alexverboon** [comment]
Adding:
"source": "playbook" 

into the watchlist body made things work for me. 
https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/new-watchlist-actions-available-for-watchlist-automation-using/bc-p/3701481#M4002

**Comment by v-amolpatil** [comment]
Hi @d3vzer0,  As per discussion with Benjamin, logic app is working, so closing this issue. Feel free to reopen if you need an assistance in it.
